src/Google.Api.Gax.Grpc/CallSettingsExtensions.cs

```diff
@@ -96,6 +97,55 @@ public static CallSettings MergedWith(this CallSettings original, CallSettings o
             settings.MergedWith(CallSettings.FromHeader(name, value));
 
         /// <summary>
+        /// Returns a <see cref="CallSettings"/> which will have an effective deadline of at least <paramref name="deadline"/>.
+        /// If <paramref name="settings"/> already observes an earlier deadline (with respect to <paramref name="clock"/>),
+        /// or if <paramref name="deadline"/> is null, the original settings will be returned.
+        /// </summary>
+        /// <param name="settings">Existing settings. May be null, meaning there are currently no settings.</param>
+        /// <param name="deadline">Deadline to enforce. May be null, meaning there is no deadline to enforce.
+        /// <param name="clock">The clock to use when computing deadlines. Must not be null.</param>
+        /// <returns>The call settings to use to observe the given deadline.</returns>
+        public static CallSettings WithEarlierDeadline(this CallSettings settings, DateTime? deadline, IClock clock)
+        {
+            GaxPreconditions.CheckNotNull(clock, nameof(clock));
+            // No deadline? Return what we already have.
+            if (deadline == null)
+            {
+                return settings;
+            }
+            // No settings, or no timing in the settings? Create a simple expiration.
+            if (settings == null || settings.Timing == null)
+            {
+                return settings.WithCallTiming(CallTiming.FromDeadline(deadline.Value));
```

Comment:
Yup, and that's fine - WithCallTiming is an extension method, and it handles the first arg being null :)